doc/modules/ensemble.rst

```diff
@@ -1,534 +1,437 @@
-in bias. In practice the variance reduction is often significant hence yielding
-an overall better model.
+  >>> X = np.array([0, 1, 2, np.nan]).reshape(-1, 1)
+  >>> y = [0, 0, 1, 1]
 
-In contrast to the original publication [B2001]_, the scikit-learn
-implementation combines classifiers by averaging their probabilistic
-prediction, instead of letting each classifier vote for a single class.
+  >>> gbdt = HistGradientBoostingClassifier(min_samples_leaf=1).fit(X, y)
+  >>> gbdt.predict(X)
+  array([0, 0, 1, 1])
 
-A competitive alternative to random forests are
-:ref:`histogram_based_gradient_boosting` (HGBT) models:
+When the missingness pattern is predictive, the splits can be done on
+whether the feature value is missing or not::
 
--  Building trees: Random forests typically rely on deep trees (that overfit
-   individually) which uses much computational resources, as they require
-   several splittings and evaluations of candidate splits. Boosting models
-   build shallow trees (that underfit individually) which are faster to fit
-   and predict.
+  >>> X = np.array([0, np.nan, 1, 2, np.nan]).reshape(-1, 1)
+  >>> y = [0, 1, 0, 0, 1]
+  >>> gbdt = HistGradientBoostingClassifier(min_samples_leaf=1,
+  ...                                       max_depth=2,
+  ...                                       learning_rate=1,
+  ...                                       max_iter=1).fit(X, y)
+  >>> gbdt.predict(X)
+  array([0, 1, 0, 0, 1])
 
--  Sequential boosting: In HGBT, the decision trees are built sequentially,
-   where each tree is trained to correct the errors made by the previous ones.
-   This allows them to iteratively improve the model's performance using
-   relatively few trees. In contrast, random forests use a majority vote to
-   predict the outcome, which can require a larger number of trees to achieve
-   the same level of accuracy.
+If no missing values were encountered for a given feature during training,
+then samples with missing values are mapped to whichever child has the most
+samples.
 
--  Efficient binning: HGBT uses an efficient binning algorithm that can handle
-   large datasets with a high number of features. The binning algorithm can
-   pre-process the data to speed up the subsequent tree construction (see
-   :ref:`Why it's faster <Why_it's_faster>`). In contrast, the scikit-learn
-   implementation of random forests does not use binning and relies on exact
-   splitting, which can be computationally expensive.
+.. _sw_hgbdt:
 
-Overall, the computational cost of HGBT versus RF depends on the specific
-characteristics of the dataset and the modeling task. It's always a good idea
-to try both models and compare their performance and computational efficiency
-on your specific problem to determine which model is the best fit.
+Sample weight support
+^^^^^^^^^^^^^^^^^^^^^
 
-.. topic:: Examples:
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor` sample support weights during
```

Comment:
```suggestion
:class:`HistGradientBoostingRegressor` support sample weights during
```
"sampling support weights" sounds like a new technique waiting to be invented :D